src/fu-engine.c

```diff
@@ -1279,22 +1357,7 @@ fu_engine_install_tasks (FuEngine *self,
 	}
 
 	/* get a new list of devices in case they replugged */
-	devices_new = g_ptr_array_new_with_free_func ((GDestroyNotify) g_object_unref);
-	for (guint i = 0; i < devices->len; i++) {
-		FuDevice *device;
-		g_autoptr(FuDevice) device_new = NULL;
-		g_autoptr(GError) error_local = NULL;
-		device = g_ptr_array_index (devices, i);
-		device_new = fu_device_list_get_by_id (self->device_list,
-						       fu_device_get_id (device),
-						       &error_local);
-		if (device_new == NULL) {
-			g_debug ("failed to find new device: %s",
-				 error_local->message);
-			continue;
-		}
-		g_ptr_array_add (devices_new, g_steal_pointer (&device_new));
-	}
+	devices_new = fu_engine_get_replugged_devices (self, devices);
```

Comment:
I do wonder if this belongs in fu-device-list.c somehow.

Reply:
I mean I think it can be dropped in fu-device-list.c if you think it's a better fit there.  Something like this:

'fu_engine_refresh_device_ids (FuDeviceList *self, GPtrArray *devices_old)`